features/bootstrap/FeatureContext.php

```diff
@@ -67,6 +67,22 @@ public function alterUserParameters(EntityScope $scope) {
     }
   }
 
+    /**
+     * Hook into user creation to test `@beforeEntityCreate`
+     *
+     * @beforeEntityCreate
+     */
+    public function alterEntityParameters(EntityScope $scope)
+    {
+        // @see `features/entity.feature`
+        // Concatenate 'First name' and 'Last name' to form user name.
+        $entity = $scope->getEntity();
+        if (isset($entity->{"First name"}) && isset($entity->{"Last name"})) {
+            $entity->name = $entity->{"First name"} . ' ' . $entity->{"Last name"};
+            unset($entity->{"First name"}, $entity->{"Last name"});
+        }
```

Comment:
This code only works for nodes, not for all entities. The code in `FeatureContext` is meant as an example to show people how things can be implemented so we should demonstrate best practices.

Instead of checking if certain parameters are set on the entity, we should check its type, I would love this code to look like this:

```
if ($entity->bundle() === 'node') {
  $entity->name = ...;
  // ...
}
```

This means however that we have to transform `$entity` from a `stdClass` into a real class. I think this would be a good thing though, and something that would be great to have in 4.x.